0019/README.md

```diff
@@ -0,0 +1,146 @@
+
+EPMMP complies with the [Inter-Service Messaging](/0003/README.md) guidelines.
+
+
+# Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+The following terms and definitions are used in this RFC.
+
+- **Endpoint metadata repository (repository)**: any service that exposes the EPMMP interface to other services for managing endpoint metadata.
+
+- **Endpoint metadata management client (client)**: any service that uses exposed EPMMP interface to manage endpoint metadata.
+
+
+# Design
+
+## Endpoint metadata get request
+
+In order to request endpoint metadata, clients MUST [emit](/0003/README.md#targeted-messaging) `EndpointMetadataGetRequest` message to the following NATS subject:
+```
+kaa.v1.service.<ep-metadata-service-instance-name>.epmmp.ep-metadata-get-request
+```
+where:
+- `<ep-metadata-service-instance-name>` is the endpoint metadata repository service instance name.
+
+The NATS message payload is an Avro object with the following schema ([0019-endpoint-metadata-get-request.avsc](./0019-endpoint-metadata-get-request.avsc)):
+
+```json
+{
+    "namespace":"org.kaaproject.ipc.epmmp.gen.v1",
+    "name":"EndpointMetadataGetRequest",
+    "type":"record",
+    "doc":"Interservice endpoint metadata get request",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services."
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds."
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"endpointId",
+            "type": "string",
+            "doc":"Identifier of the endpoint, on behalf of which metadata is requested."
+        },
+        {
+            "name":"include",
+            "type":{
+                "type":"array",
+                "items":"string"
+            },
+            "default":[],
+            "doc":"List of metadata fields. If not specified all fields are icluded."
+        }
+    ]
+}
+```
+
+Client MUST specify `replyTo` NATS subject in the emit message according to the [session affinity design](/0003/README.md#session-affinity).
+
+
+## Endpoint metadata response
+
+To receive a response, the client MUST be subscribed to the replyTo subject specified in the request message.
+
+The NATS message payload is an Avro object with the following schema ([0019-endpoint-metadata.avsc](./0019-endpoint-metadata.avsc)):
```

Comment:
*Endpoint metadata get response* message MUST be sent by repository in response to an [Endpoint metadata get  request message](#endpoint-metadata-get-request).
Repository MUST publish endpoint metadata get response message to the subject provided in the NATS `replyTo` field of the request.

Endpoint metadata get response message payload MUST be an Avro-encoded object with the following schema ([0019-endpoint-metadata.avsc](./0019-endpoint-metadata.avsc)):